### PR TITLE
Clarifai LangChain improvements and embeddings

### PR DESCRIPTION
**Pull request by zeiler**
This PR improves upon the Clarifai LangChain integration with improved docs, errors, args and the addition of embedding model support in LancChain for Clarifai's embedding models and an overview of the various ways you can integrate with Clarifai added to the docs.

@hwchase17 please take a look.

Twitter: @clarifai

### CODE REVIEW

File: langchain/llms/clarifai.py [hunk 1]
```diff
@@ -13,92 +13,107 @@
...
-
-    def _call(
-        self,
-        prompt: str,
-        stop: Optional[List[str]] = None,
-        run_manager: Optional[CallbackManagerForLLMRun] = None,
-        **kwargs: Any
-    ) -> str:
-        """Call out to Clarfai's PostModelOutputs endpoint.
+    values["pat"] = get_from_dict_or_env(values, "pat", "CLARIFAI_PAT")
```
Comment (baskaryan) [inline langchain/llms/clarifai.py:59]: this is a breaking change, are we sure we want to make it?

Comment (zeiler) [reply langchain/llms/clarifai.py]: yeah, the previous one wasn't aligned with how our python client normally takes in the PAT like the ClarifaiAuthHelper class used within here, so wanted to clean that up while the integration is still new.

**Comment by baskaryan** [comment]
lgtm, thanks @zeiler!

**Comment by baskaryan** [comment]
fixed some linting/testing errors and landed in #7324!

**Pull request closed without merging**